### PR TITLE
tox3: Support provision of tox 4 with the min_version option

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/tox-dev/tox/issues/2661

# Thanks for contribution

- [x] ran the linter to address style issues (`tox -e fix`)
- [x] wrote descriptive pull request text
- [x] ensured there are test(s) validating the fix
- [x] added news fragment in `docs/changelog` folder
- [x] updated/extended the documentation

### CODE REVIEW

**Comment by hroncok** [comment]
The new test needs to be skipped on old Pythons.

File: tests/unit/session/test_provision.py [hunk 1]
```diff
@@ -27,14 +27,21 @@ def next_tox_major():
...
+
+
+def test_provision_min_version_is_requires(newconfig, minversion_option, next_tox_major):
     with pytest.raises(MissingRequirement) as context:
         newconfig(
             [],
             """\
             [tox]
-            minversion = {}
+            {} = {}
```
Comment (hroncok) [inline tests/unit/session/test_provision.py:42]: I wonder if adding a new fixture that evaluates to this expression is worth it or not.

File: tests/integration/test_provision_int.py [hunk 2]
```diff
@@ -83,7 +108,7 @@ def test_provision_interrupt_child(initproj, monkeypatch, capfd, signal_type):
             "tox.ini": """
                     [tox]
                     skipsdist=True
-                    minversion = 3.7.0
+                    minversion = 3.7.0,<4
```
Comment (hroncok) [inline tests/integration/test_provision_int.py:111]: I don't know if this is explicitly supported but thankfully it works. Without it the tests were failing (as they were provisioning tox 4).

**Comment by hroncok** [comment]
I'm not sure if the Windows failures are related or not. I don't have a Windows development environment to reproduce.

**Comment by gaborbernat** [comment]
> I'm not sure if the Windows failures are related or not. I don't have a Windows development environment to reproduce.

Can't accept the PR without passing CI so would need to figure it out 👍

**Comment by hroncok** [comment]
Attempting to solve the `UnicodeDecodeError: 'charmap' codec can't decode byte 0x8f in position 164: character maps to <undefined>` thing by reading files as utf-8.

**Comment by hroncok** [comment]
Adding encoding=utf-8 to subprocess.check_output in the failing test seems to fix the problem however Python 3.5 does not support that parameter. I could try working around that by monkeypatching locale.getpreferredencoding to return utf-8 or by skipping the test on Python 3.5. Honestly, I have no idea what is going on here and why did this test not fail previously.

**Comment by hroncok** [comment]
I went too far trying to bend the code for Python 3.5 and 2.7 and I completely forgot I can run subprocess on bytes and decode the output explicitly. Hopefully that'll work.

**Comment by hroncok** [comment]
Now the coverage job failed with:

>  Could not find the branch to compare to. Does 'origin/master' exist?

Added another commit.

File: tests/integration/test_provision_int.py [hunk 3]
```diff
@@ -68,6 +68,31 @@ def test_provision_from_pyvenv(initproj, cmd, monkeypatch):
...
+def test_provision_tox_4(initproj, cmd, monkeypatch):
+    initproj(
+        "pkg123-0.7",
+        filedefs={
+            "tox.ini": """\
+                [tox]
+                no_package=True
+                min_version = 4
+                [testenv]
+                commands=python -c "import os; print('assert this')"
```
Comment (hroncok) [inline tests/integration/test_provision_int.py:84]: The import os part is a leftover from debugging a d should have been removed. Sorry about that.

**Comment by hroncok** [comment]
Thanks for merging.

**Pull request merged**